### ISSUE TITLE
release the code?

### ISSUE THREAD

**Issue by fuxuliu** [HanselYu/PreKcat#1]
Hi, yu.

I noticed your prekcat paper, which is a great paper. 
May I ask when you will release the code?

**Comment by HanselYu** [comment]
Thansks for your attention! Because some details need to be optimized, the code is expected to be released from February to March.

**Comment by subhansb8** [comment]
Hi, 
Thats a great paper compare to DLKcat.
I wonder if the code could be available.

**Comment by jiejiangtao** [comment]
Hi,
I wonder if the code could be available.

**Comment by HanselYu** [comment]
Thansks for all of your attention!
We have released a new version of PreKcat, the code is as follow:
https://github.com/Luo-SynBioLab/UniKP
I hope this may be clearer and helpful.

**Comment by jiejiangtao** [comment]
Thank you very much for your help.&nbsp;
I have also trained myself through these two pre-trained models and xgboost, but the R2 value on my test set is only 0.62, which is far less accurate than yours. My data set is divided into training set, test set and valid set, and the ratio is 8:1:&nbsp;1. I also looked at the DLkcat code and the data set is divided this way, but it seems to me that your code is divided only into the training set and the test set, the ratio is 9:1.&nbsp;I want to know if my test set R2 is low because of that, or if I'm missing something in your code。
I see the training part of your code is this, I see that you divide the training set and the test set, or I misunderstood the meaning of this part of the code.





解江涛
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "HanselYu/PreKcat"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年10月18日(星期三) 晚上10:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [HanselYu/PreKcat] release the code? (Issue #1)





 
Thansks for all of your attention!
 We have released a new version of PreKcat, the code is as follow:
 https://github.com/Luo-SynBioLab/UniKP
 I hope this may be clearer and helpful.

**Comment by HanselYu** [comment]
I suggest you try Extra Trees as the head model, our experience proves that it is currently the best.
For the comparison, since we have not done additional parameter optimization here, we directly compared the performance with the test set. 
Additionally, the performance may be a little different due to different dataset division, but in overall,  after randomly mutiple divisions, the performance should be consistent.

**Comment by jiejiangtao** [comment]
Thank you for your reply, but I still don't understand why the final effect is the same when the data sets are divided differently.

Although the proportion of the test set is 10%, but I see your code in the training set is 90%, and dlkcat in the training set is 80%, in theory, the result should be better if the training set accounts for more.&nbsp;If you want to compare the results with dlkcat, then the training set should also account for 80% of the data set division.





解江涛
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "HanselYu/PreKcat"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年10月19日(星期四) 下午3:04
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [HanselYu/PreKcat] release the code? (Issue #1)





 
I suggest you try Extra Trees as the head model, our experience proves that it is currently the best.
 For the comparison, since we have not done additional parameter optimization here, we directly compared the performance with the test set.
 Additionally, the performance may be a little different due to different dataset division, but in overall,  after randomly mutiple divisions, the performance should be consistent.

**Comment by HanselYu** [comment]
I apologize for any confusion earlier.
What I meant is that the overall evaluation performance should remain consistent when the data is randomly divided multiple times, with the training, validation, and test set ratios set at 8:1:1. 
I greatly appreciate your input, and in theory, having an 80% training set could be a fairer conmparison.
We will consider leave 10% of the dataset to optimize the model parameters.
